### PR TITLE
Feature/zcs 11481

### PR DESCRIPTION
**Pull request by vishnukantjangid**
**Feature**
Need API for scheduling HSM policy execution.

**Solution**
Created API for scheduling HSM policy and also created a script **zmschedulesmpolicy** for schedule from command line.

**Options for zmschedulesmpolicy**

**-l: list** - print existing schedule
**-h: help** - get help about zmschedulesmpolicy (default command)
**-f: flush** - remove current schedule (cancel all scheduled Storage Management policy)
**-e: edit** - edit current schedule with specified schedule

### CODE REVIEW

File: soap/src/java/com/zimbra/soap/admin/message/ScheduleSMPolicyRequest.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    public ScheduleSMPolicyRequest(Name server) {
+        this.server = server;
+    }
+
+    public Name getServer() { return server; }
+
+    public MoreObjects.ToStringHelper addToStringInfo(
+                MoreObjects.ToStringHelper helper) {
+        return helper
+            .add("server", server);
```
Comment (dasiyogesh) [inline soap/src/java/com/zimbra/soap/admin/message/ScheduleSMPolicyRequest.java:64]: move it to above line

File: soap/src/java/com/zimbra/soap/admin/message/ScheduleSMPolicyRequest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+import com.zimbra.common.soap.AdminConstants;
+import com.zimbra.common.soap.BackupConstants;
+import com.zimbra.common.soap.HsmConstants;
+import com.zimbra.soap.admin.type.Name;
+
+/**
+ * @zm-api-command-network-edition
+ * @zm-api-command-auth-required true
+ * @zm-api-command-admin-auth-required true
```
Comment (RakeshAMore) [inline soap/src/java/com/zimbra/soap/admin/message/ScheduleSMPolicyRequest.java:34]: Correct these statements, I see "Schedule backup" its schedule policy :)

File: soap/src/java/com/zimbra/soap/admin/message/ScheduleSMPolicyResponse.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    public ScheduleSMPolicyResponse(boolean smSchedulePolicyEnabled) {
+        this.smSchedulePolicyEnabled = ZmBoolean.fromBool(smSchedulePolicyEnabled);
+    }
+
+    public void setError(String error) { this.error = error; }
+    public void setSmScheduleStartTime(Integer smSchedulePolicyStartTime) {
+        this.smSchedulePolicyStartTime = smSchedulePolicyStartTime;
+    }
+
+    public boolean getSmScheduleEnabled() { return ZmBoolean.toBool(smSchedulePolicyEnabled); }
```
Comment (RakeshAMore) [inline soap/src/java/com/zimbra/soap/admin/message/ScheduleSMPolicyResponse.java:72]: Even if it's a single statement method, keep it three-liner. hope you go what I am saying :) 
``` 
public boolean getSmScheduleEnabled() { 
   return ZmBoolean.toBool(smSchedulePolicyEnabled);
}
```
Please use the same pattern for all single line methods.. It makes code more readable.

File: soap/src/java/com/zimbra/soap/admin/message/ScheduleSMPolicyResponse.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                MoreObjects.ToStringHelper helper) {
+        return helper
+            .add("smSchedulePolicyEnabled", smSchedulePolicyEnabled)
+            .add("smSchedulePolicyStartTime", smSchedulePolicyStartTime)
+            .add("error", error);
+    }
+
+    @Override
+    public String toString() {
+        return addToStringInfo(MoreObjects.toStringHelper(this))
```
Comment (shrutig0510) [inline soap/src/java/com/zimbra/soap/admin/message/ScheduleSMPolicyResponse.java:96]: indentation issue

**Review by zimsuchitgupta — APPROVED** [review]

**Review by shrutig0510 — APPROVED** [review]

**Review by deepakzimb — APPROVED** [review]

**Review by abhishekdeshmukh2209 — APPROVED** [review]

**Pull request merged**